src/qibolab/platform.py

```diff
@@ -214,22 +169,49 @@ def disconnect(self):
                 instrument.disconnect()
         self.is_connected = False
 
-    def _execute(self, sequence, options, **kwargs):
+    def _apply_config_updates(
+        self, updates: list[dict[str, Config]]
+    ) -> dict[str, Config]:
+        """Apply given list of config updates to the default configuration and
+        return the updated config dict.
+
+        Args:
+            updates: list of updates, where each entry is a dict mapping component name to new config. Later entries
+                     in the list override earlier entries (if they happen to update the same thing).
+        """
+        components = self.components.copy()
```

Comment:
I am not sure at what stage of the draft was this commented, but currently the component configs dict in the platform is never updated. The intention is
1. updates will happen only during execution, and platform always stays at defaults.
2. if users want to dump updated configs, there is a function that does this by accepting the same updates again. This is supposed to replace the current use case of qibocal, where they mutate the platform itself and then dump it. And now I remember that I have forgotten to implement this point :D

Reply:
> I am not sure at what stage of the draft was this commented,

Again, nothing changed since then, the line is still there:
https://github.com/qiboteam/qibolab/blob/982c843f41630f26a926af7ae2b38e720eb5f6dd/src/qibolab/platform/platform.py#L190

> updates will happen only during execution, and platform always stays at defaults.

I don't care much about the platform being the same or mutating. There is almost always a single platform around for each process, it is almost a singleton. And it's updated in very predictable steps, even in Qibocal.

However, even fine if you do not want to update the platform. But I want at least to be able to update a components' dictionary. In-place, without copying.
Then, if I want to copy I will, but that's an extra operation.
If I have 1M components, and I want to update one, here I'm forced to copy 1M, though is definitely not necessary for what is done by the platform.

So, if you do not want to mutate the platform, split the updating function as:
```py
def configs_updates(components, updates):
    ... # no copy
    return updated_components

class Platform:
    ...

    # if you really still need this...
    def _apply_config_updates(self, updates):
        components = self.components.copy()
        return configs_updates(components, updates)
```

Reply:
Sure, I will keep an eye on this when we have 1M components.